Spatial.gd

```diff
@@ -632,31 +638,18 @@ func gen_new_icon(position: Vector3, texture_path: String, waypoint_icon, catego
 	var category_data = category_item.get_metadata(0)
 	category_data.category3d.add_icon(new_icon)
 
-# This function take all of the currently rendered objects and converts it into
-# the data format that is saved/loaded from.
-func data_from_renderview():
```

Comment:
If we are no longer relying on the internal state of the nodes how can we be sure that the state of the nodes is in sync with whatever we are saving

Reply:
We should be relying on the proto as the source of truth. So Icon.gd and Route.gd should now be set to update the proto any time a change is made to the node.